### ISSUE TITLE
Gstreamer playback problem "intel_do_flush_locked failed: No such file or directory"

### ISSUE THREAD

**Issue by ilmaisin** [UnitedRPMs/libva-intel-driver#1]
I am unable to play h.264 videos with Totem, gstreamer codecs from United RPMs and the integrated GPU of an Intel i3 4130T.

Reproduction:

```
# dnf install gstreamer{1,}-{ffmpeg,libav,plugins-{good,ugly,bad{,-free,-nonfree}}} --setopt=strict=0
# dnf install libva-intel-driver
$ totem toystory.mp4
```

Console output of Totem under X11:

```
DtsGetHWFeatures: Create File Failed
DtsGetHWFeatures: Create File Failed
Running DIL (3.22.0) Version
DtsDeviceOpen: Opening HW in mode 0
DtsDeviceOpen: Create File Failed
libva info: VA-API version 0.39.2
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib64/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_0_39
libva info: va_openDriver() returns 0
intel_do_flush_locked failed: No such file or directory
```

On X11, the app crashes immediately. On Wayland, the audio stops after a few seconds and the video does not show at all, only a black screen is shown.

If I remove package libva-intel-driver but keep gstreamer1-vaapi, it fails probably because there is no driver. The output is following:

```
DtsGetHWFeatures: Create File Failed
DtsGetHWFeatures: Create File Failed
Running DIL (3.22.0) Version
DtsDeviceOpen: Opening HW in mode 0
DtsDeviceOpen: Create File Failed
libva info: VA-API version 0.39.2
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib64/dri/i965_drv_video.so
libva info: va_openDriver() returns -1
libva info: VA-API version 0.39.2
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib64/dri/i965_drv_video.so
libva info: va_openDriver() returns -1
libva info: VA-API version 0.39.2
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib64/dri/i965_drv_video.so
libva info: va_openDriver() returns -1
libva info: VA-API version 0.39.2
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib64/dri/i965_drv_video.so
libva info: va_openDriver() returns -1
libva info: VA-API version 0.39.2
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib64/dri/i965_drv_video.so
libva info: va_openDriver() returns -1
libva info: VA-API version 0.39.2
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib64/dri/i965_drv_video.so
libva info: va_openDriver() returns -1
libva info: VA-API version 0.39.2
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib64/dri/i965_drv_video.so
libva info: va_openDriver() returns -1
```

I am not fully sure if the problem is with this package, the codecs themselves or Fedora's `gstreamer1-vaapi`. The same problem is present with RPM Fusion's packages.

If I remove package `gstreamer1-vaapi`, the video plays, but of course without hardware acceleration.

Test video used is this: http://www.html5videoplayer.net/videos/toystory.mp4

Bug report to Fedora against `gstreamer1-vaapi`: https://bugzilla.redhat.com/show_bug.cgi?id=1354123

Bug report to RPMFusion: https://bugzilla.rpmfusion.org/show_bug.cgi?id=4113

**Comment by paulcarroty** [comment]
@ilmaisin thanks for report. Can you test HW acceleration in mpv/gnome-mpv with the same video? Cast @sergiomb2 @kuboosoft

**Comment by ilmaisin** [comment]
It works fine with mpv. Console output:

```
$ mpv --hwdec=vaapi toystory.mp4
Playing: toystory.mp4
 (+) Video --vid=1 (*) (h264)
 (+) Audio --aid=1 --alang=und (*) (aac)
libva info: VA-API version 0.39.2
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib64/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_0_39
libva info: va_openDriver() returns 0
Using hardware decoding (vaapi).
VO: [opengl] 640x360 vaapi
AO: [pulse] 44100Hz stereo 2ch float
AV: 00:00:28 / 00:02:29 (19%) A-V:  0.000


Exiting... (Quit)
```

**Comment by paulcarroty** [comment]
@ilmaisin the same, tested on Sandy/Ivy. There is the problem with Intel GPU driver:

https://bugzilla.redhat.com/show_bug.cgi?id=1309446
https://bugs.freedesktop.org/show_bug.cgi?id=71759

Waiting for upstream fix. You can use temporally a (gnome-)mpv for video playback.

**Comment by paulcarroty** [comment]
@kuboosoft can you HW acc with gstreamer-1.9?

**Comment by ilmaisin** [comment]
@paulcarroty, that freedesktop bug is from 2013. I am certain it worked on F23.

**Comment by ilmaisin** [comment]
Anyway, running `export LIBGL_DRI3_DISABLE=1` before `totem` seems to fix it, so it is most probably that bug.

**Comment by paulcarroty** [comment]
@ilmaisin all last post from 2016. It's definitely a GPU driver bug. 

So, the patch is available, you can rebuild driver in Copr: https://bugs.freedesktop.org/attachment.cgi?id=122880

**Comment by ilmaisin** [comment]
@paulcarroty, I am not software developer but could probably eventually do it. Would it help getting it patched in United RPMs and/or RPMFusion?

Is the Intel driver by the way allowed in COPR? I am asking because it is not distributed by Fedora itself for some reason.

Edit: Oops, I noticed you obviously meant the mesa driver and not this driver. There should not be distributability problems in that.

**Comment by paulcarroty** [comment]
> Is the Intel driver by the way allowed in COPR?

@ilmaisin yep, all Fedora packages is allowed in Copr.

Add patch it's easy:

```
# dnf download --source xorg-x11-drv-intel

$ rpm -i *.src.rpm

$ vi ~/rpmbuild/SPECS/*spec
```

Include your patch: https://fedoraproject.org/wiki/Packaging:Guidelines#Patch_Guidelines

Rebuild your package and upload to Copr:

```
rpmbuild -bs ~/rpmbuild/SPECS/*spec
```

**Comment by ilmaisin** [comment]
I am experiencing a build failure on copr. It looks like as if Xorg developement packages were not installed or something. I don't know how to proceed.

https://copr-be.cloud.fedoraproject.org/results/ilmaisin/xorg-x11-drv-intel-dri3-patch/fedora-24-x86_64/00376635-xorg-x11-drv-intel/build.log.gz

```
Mock Version: 1.2.18
ENTER ['do'](['bash', '--login', '-c', '/usr/bin/rpmbuild -bs --target x86_64 --nodeps /builddir/build/SPECS/xorg-x11-drv-intel.spec'], printOutput=Falsegid=135logger=<mockbuild.trace_decorator.getLog object at 0x7fd57a5d4978>env={'LANG': 'en_US.UTF-8', 'TERM': 'vt100', 'PATH': '/usr/bin:/bin:/usr/sbin:/sbin', 'SHELL': '/bin/bash', 'PS1': '<mock-chroot> \\s-\\v\\$ ', 'HOSTNAME': 'mock', 'HOME': '/builddir', 'PROMPT_COMMAND': 'printf "\\033]0;<mock-chroot>\\007"'}uid=1001timeout=0chrootPath='/var/lib/mock/fedora-24-x86_64-mockbuilder-6204/root'user='mockbuild'shell=False)
Executing command: ['bash', '--login', '-c', '/usr/bin/rpmbuild -bs --target x86_64 --nodeps /builddir/build/SPECS/xorg-x11-drv-intel.spec'] with env {'LANG': 'en_US.UTF-8', 'TERM': 'vt100', 'PATH': '/usr/bin:/bin:/usr/sbin:/sbin', 'SHELL': '/bin/bash', 'PS1': '<mock-chroot> \\s-\\v\\$ ', 'HOSTNAME': 'mock', 'HOME': '/builddir', 'PROMPT_COMMAND': 'printf "\\033]0;<mock-chroot>\\007"'} and shell False
sh: xserver-sdk-abi-requires: command not found
sh: xserver-sdk-abi-requires: command not found
Package xorg-server was not found in the pkg-config search path.
Perhaps you should add the directory containing `xorg-server.pc'
to the PKG_CONFIG_PATH environment variable
No package 'xorg-server' found
warning: Could not canonicalize hostname: copr-builder-302864523.novalocal
Building target platforms: x86_64
Building for target x86_64
Wrote: /builddir/build/SRPMS/xorg-x11-drv-intel-2.99.917-23.20160512.fc24.ilmaisin1.src.rpm
Child return code was: 0
ENTER ['do'](['bash', '--login', '-c', '/usr/bin/rpmbuild -bb --target x86_64 --nodeps /builddir/build/SPECS/xorg-x11-drv-intel.spec'], printOutput=Falseprivate_network=Truelogger=<mockbuild.trace_decorator.getLog object at 0x7fd57a5d4978>gid=135env={'LANG': 'en_US.UTF-8', 'TERM': 'vt100', 'PATH': '/usr/bin:/bin:/usr/sbin:/sbin', 'SHELL': '/bin/bash', 'PS1': '<mock-chroot> \\s-\\v\\$ ', 'HOSTNAME': 'mock', 'HOME': '/builddir', 'PROMPT_COMMAND': 'printf "\\033]0;<mock-chroot>\\007"'}uid=1001timeout=0chrootPath='/var/lib/mock/fedora-24-x86_64-mockbuilder-6204/root'user='mockbuild'shell=False)
Executing command: ['bash', '--login', '-c', '/usr/bin/rpmbuild -bb --target x86_64 --nodeps /builddir/build/SPECS/xorg-x11-drv-intel.spec'] with env {'LANG': 'en_US.UTF-8', 'TERM': 'vt100', 'PATH': '/usr/bin:/bin:/usr/sbin:/sbin', 'SHELL': '/bin/bash', 'PS1': '<mock-chroot> \\s-\\v\\$ ', 'HOSTNAME': 'mock', 'HOME': '/builddir', 'PROMPT_COMMAND': 'printf "\\033]0;<mock-chroot>\\007"'} and shell False
Building target platforms: x86_64
Building for target x86_64
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.Ct0Qmj
+ umask 022
+ cd /builddir/build/BUILD
+ cd /builddir/build/BUILD
+ rm -rf xf86-video-intel-20160512
+ /usr/bin/bzip2 -dc /builddir/build/SOURCES/xf86-video-intel-20160512.tar.bz2
+ /usr/bin/tar -xof -
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ /usr/bin/bzip2 -dc /builddir/build/SOURCES/intel-gpu-tools-1.9.tar.bz2
+ /usr/bin/tar -xof -
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd xf86-video-intel-20160512
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ echo 'Patch #0 (intel-gcc-pr65873.patch):'
Patch #0 (intel-gcc-pr65873.patch):
+ /usr/bin/patch -p1 -b --suffix .gcc --fuzz=0
patching file src/sna/compiler.h
+ echo 'Patch #2 (0001-sna-Let-modestting-glamor-handle-gen9.patch):'
Patch #2 (0001-sna-Let-modestting-glamor-handle-gen9.patch):
+ /usr/bin/patch -p1 --fuzz=0
patching file src/intel_module.c
+ echo 'Patch #3 (0002-Fix-fd-and-mem-leak-when-intel_scrn_create-fails.patch):'
Patch #3 (0002-Fix-fd-and-mem-leak-when-intel_scrn_create-fails.patch):
+ /usr/bin/patch -p1 --fuzz=0
patching file src/intel_device.c
patching file src/intel_driver.h
patching file src/intel_module.c
+ echo 'Patch #3 (0002-Fix-fd-and-mem-leak-when-intel_scrn_create-fails.patch):'
Patch #3 (0002-Fix-fd-and-mem-leak-when-intel_scrn_create-fails.patch):
+ /usr/bin/patch -p1 --fuzz=0
patching file src/intel_device.c
Reversed (or previously applied) patch detected!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file src/intel_device.c.rej
patching file src/intel_driver.h
Reversed (or previously applied) patch detected!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file src/intel_driver.h.rej
patching file src/intel_module.c
Reversed (or previously applied) patch detected!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
4 out of 4 hunks ignored -- saving rejects to file src/intel_module.c.rej
error: Bad exit status from /var/tmp/rpm-tmp.Ct0Qmj (%prep)
RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.Ct0Qmj (%prep)
Child return code was: 1
EXCEPTION: [Error()]
Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/mockbuild/trace_decorator.py", line 88, in trace
    result = func(*args, **kw)
  File "/usr/lib/python3.4/site-packages/mockbuild/util.py", line 551, in do
    raise exception.Error("Command failed. See logs for output.\n # %s" % (command,), child.returncode)
mockbuild.exception.Error: Command failed. See logs for output.
 # bash --login -c /usr/bin/rpmbuild -bb --target x86_64 --nodeps /builddir/build/SPECS/xorg-x11-drv-intel.spec
```

**Comment by paulcarroty** [comment]
@ilmaisin the patch is bad for current version, I'm sorry.

**Comment by kuboosoft** [comment]
It is a problem with DRI3;  I do not have DRI3 support with my old video card... i915 Intel. But I play fine all videos with totem, vlc, mpv (all)

**Comment by paulcarroty** [comment]
@kuboosoft new logs at F26 and new libva-intel-driver 1.7.2: Internal data stream error

```
(totem:17079): Gtk-WARNING **: GtkAlignment 0x55e3947dd350 is drawn without a current allocation. This should not happen.

(totem:17079): Gtk-WARNING **: GtkBox 0x55e3947a0750 is drawn without a current allocation. This should not happen.

(totem:17079): Gtk-WARNING **: GtkImage 0x55e3947cc2b0 is drawn without a current allocation. This should not happen.
libva info: VA-API version 0.39.2
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib64/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_0_39
libva info: va_openDriver() returns 0
libva info: VA-API version 0.39.2
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib64/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_0_39
libva info: va_openDriver() returns 0

(totem:17079): Gtk-WARNING **: GtkClutterOffscreen 0x7f4754011f70 is drawn without a current allocation. This should not happen.

(totem:17079): Gtk-WARNING **: GtkToolbar 0x55e394362ab0 is drawn without a current allocation. This should not happen.

(totem:17079): Gtk-WARNING **: GtkToolItem 0x55e3944281a0 is drawn without a current allocation. This should not happen.

(totem:17079): Gtk-WARNING **: GtkBox 0x55e3944c4140 is drawn without a current allocation. This should not happen.
...
```

Looks like HW ACC is fixed, we should close this bug and open another. Anybody can accept it on F24?

**Comment by ilmaisin** [comment]
Weird. According to the [Freedesktop bug 71759](https://bugs.freedesktop.org/show_bug.cgi?id=71759) discussion and the discussion on the mesa-dev mailing list (search for  `i965: import prime buffers in the current context not screen`) the problem has been quite conclusively traced to the DRI driver. However it is possible that the new libva-intel-driver uses some API commands differently or something.

**Comment by paulcarroty** [comment]
Works in F26, anyone can reproduce it?

**Comment by ilmaisin** [comment]
@paulcarroty Sorry, I have forgot this. It has been fixed a long time ago, see the freedesktop bug link above.

**Comment by sergiomb2** [comment]
from my comment in https://bugzilla.redhat.com/show_bug.cgi?id=1309446#c6
`dnf remove gstreamer1-vaapi ` should fix it , also was declared fixed with Mesa-12.0.3-2.

**Comment by paulcarroty** [comment]
@sergiomb2 HW acceleration now works, I've tested it with totem.

![screenshot from 2016-12-29 17-26-55](https://cloud.githubusercontent.com/assets/1550705/21547024/1004ee96-cdec-11e6-96bb-776e3d3ae3e7.png)
